web/src/engine/keyboard/src/kmxPlus/keys.ts

```diff
@@ -0,0 +1,185 @@
+import { List } from './list.js';
+import { Strs } from './strs.js';
+import { decodeNumber } from './utils.js';
+
+type Flick = { dirSequence: readonly string[], keyId: string };
+type FlickSet = { flicks: Flick[], flickId: string };
+type VkeyMapping = { code: number, modifiers: number, keyIndex: number };
+export type KeySpec = {
+  to: string;
+  isGap: boolean;
+  id: string;
+  switch: string;
+  width: number;
+  longpress: readonly string[];
+  defaultLongpressId: string;
+  multitap: readonly string[];
+  flicks: Flick[];
+  keyCode: number,
+  modifiers: number
+};
+
+export class Keys {
+  readonly name = 'keys';
+  readonly data: Uint8Array;
+  readonly keys: Readonly<Map<string, KeySpec>>;
+
+  constructor(rawData: Uint8Array, strs: Strs, list: List) {
+    this.data = rawData;
+
+    const keyCount    = decodeNumber(rawData,  8);
+    const flicksCount = decodeNumber(rawData, 12);
+    const flickCount  = decodeNumber(rawData, 16);
+
+    const FLICK_GROUP_LEN = 12;
+    const FLICK_LEN = 8;
+
+    const keyTableOffset = 24;
+    const flicksTableOffset = keyTableOffset + keyCount * 36; // 9 entries * 4 bytes ea
+    const flickTableOffset  = flicksTableOffset + flicksCount * FLICK_GROUP_LEN;
+    const kmapTableOffset = flickTableOffset + flickCount * FLICK_LEN;
+
+    // keys.keys subtable wants to index within keys.flicks subtable.
+    // So let's build the latter first.
+
+    const flickSets = this.processFlickSubtables(strs, list, keyTableOffset);
+    const vkeyMap = this.processKeymap(kmapTableOffset);
+    const keybag = this.processKeybag(strs, list, flickSets, vkeyMap);
+
+    const keyMap: Map<string, KeySpec> = new Map();
+    for(let key of keybag) {
+      keyMap.set(key.id, key);
+    }
+    this.keys = keyMap;
+  }
+
+  private processFlickSubtables(strs: Strs, list: List, baseOffset: number) {
+    const rawData = this.data;
+
+    const keyCount    = decodeNumber(rawData,  8);
+    const flicksCount = decodeNumber(rawData, 12);
+
+    const FLICK_GROUP_LEN = 12;
```

Comment:
now, shouldn't this file be able to use https://github.com/keymanapp/keyman/blob/master/core/include/ldml/keyman_core_ldml.ts ? 
so:

```suggestion
    const FLICK_GROUP_LEN = Constants.length_keys_flick_list;
```